src/main/java/org/gwtbootstrap3/extras/toggleswitch/client/ui/base/ToggleSwitchBase.java

```diff
@@ -193,11 +193,14 @@ public HandlerRegistration addValueChangeHandler(final ValueChangeHandler<Boolea
         return addHandler(handler, ValueChangeEvent.getType());
     }
 
+    /**
+     * Gets the value of this widget.
+     * 
+     * return the value of the undelying {@link CheckBox}.  This is important because the editor's 
+     * {@link #getValue()} method often is called when not attached.
+     */
     @Override
     public Boolean getValue() {
```

Comment:
This should be the main change of your commit. And I suggest to add a big FIXME here to reference the native JS issue. Once this is fixed, we should back to the original implementation.